### ISSUE TITLE
Tasker Support

### ISSUE THREAD

**Issue by srijansaxena11** [AmazMod/AmazMod#907]
Support for Tasker and similar apps will be great and highly useful for someone like me. Similar support exists in Notify and Fitness app but for MI Band 2 and Amazfit Bip only. The support may include sending intents when the person falls asleep, wakes up and things like that. Also it may include actions to be performed on Amazfit such as turning WiFi on/off, silent mode, etc. Basically everything possible through app via Tasker so that it can be combined with other triggers and do useful stuff. Mark this as a feature request or whatever you feel correct for it. I am hereby attaching the Notify and Fitness Tasker support page: http://www.mibandnotify.com/help/tasker_send_intent.php

Thank you and keep up the good work with AmazMod.

**Comment by micrusa** [comment]
This would be really useful, and there are a lot of things that can be added, however we can't get sleep data before sleep has finished with stock tracker, about the wifi on/off and silent mode they can be added easily, it would be just listening to tasker intents and send it to the watch the same way we already do it

**Comment by srijansaxena11** [comment]
So, the mechanism is different from Amazfit Bip and other MI Band wearables? We cannot detect when we fall asleep? I mean when sleep tracking actually started? This is weird. I mean there should be a trigger point for the official App itself in order to calculate the sleep, isn't it? And I don't understand what do you mean by stock tracker. Stock sleep tracking algorithm? I am glad that actions can be implemented. You can just provide us intents name like Notify and Fitness does and we can set it up in Tasker.

**Comment by micrusa** [comment]
> So, the mechanism is different from Amazfit Bip and other MI Band wearables? We cannot detect when we fall asleep? I mean when sleep tracking actually started? This is weird. I mean there should be a trigger point for the official App itself in order to calculate the sleep, isn't it? And I don't understand what do you mean by stock tracker. Stock sleep tracking algorithm? I am glad that actions can be implemented. You can just provide us intents name like Notify and Fitness does and we can set it up in Tasker.

Amazfits without android and mi bands are completely different, I read sleep code to try to create smart alarms and iirc all sleep data wasn't saved until sleep was finished, all data is just in the watch app until it's saved. And yes, by stock tracker I mean stock sleep tracking, I'll dig more into code and see if it's possible but I don't think so

**Comment by srijansaxena11** [comment]
Yeah thats also a nice idea. You can implement Tasker actions for AmazMod and sleep triggering will be handled by Sleep as Android.

**Comment by fruchttiger87** [comment]
I would love tasker Integration too.
If there was a possibility to trigger an event from the watch you could so nearly everything with the amazfit watch when it is comnected to the phone:
control smarthome devices
pull information from the phone to the watch
I played a little bit with tasker and autonotification. I was able to send images from a whatsapp message to my Verge.
So Tasker only needs a trigger. If we could get the possibility to send one from the watch er have endless possibilities.

But before I forget: Really nice app! You die a really good job. Thank you for that!
